include/exadg/fluid_structure_interaction/precice/driver_fluid.h

```diff
@@ -218,7 +216,32 @@ class DriverFluid : public Driver<dim, Number>
 
       // post-solve
       if(is_new_time_window)
+      {
+        // computes new time-step size
         fluid->time_integrator->advance_one_timestep_post_solve();
+        // next, we synchronize the time-step sizes. Subcycling would in theory be compatible in
+        // explicit coupling schemes here. In implicit coupling schemes, we need matching
+        // time-window sizes (either constant (serial or parallel schemes) or adaptively (serial
+        // scheme)) as the time-step size push back happens in the
+        // 'advance_one_timestep_post_solve()' and we cannot change two subsequent time-step sizes
+        // without a push back operation, otherwise we falsify the time integrator. However, the
+        // treatment of the boundary conditions is not handled correctly when using subcycling, as
+        // we apply the 'whole displacement' within the first time-step. In order to cope with
+        // subcycling, either the boundary condition needs to be adopted or we need to wait for a
+        // newer preCICE version (currently v2.3.0) which can handle this.
+        // Hence, we do not adjust the time-step-size of the fluid solver, but rather Assert that it
+        // is still valid. Note that we would run into a preCICE error otherwise reporting the same
+        // issue otherwise.
+        // fluid->time_integrator->set_current_time_step_size(
+        //   std::min(this->allowed_time_step_size, fluid->time_integrator->get_time_step_size()));
+        Assert(
+          fluid->time_integrator->get_time_step_size() <
+            this->allowed_time_step_size + std::numeric_limits<double>::min(),
+          dealii::ExcMessage(
+            "The solver time-step size exceeded the maximum admissible time-step size allowed by preCICE. "
+            "If you select adaptive time-stepping, make sure to let the Fluid participant steer the time-window size. "
```

Comment:
How do I make this decision ("let the Fluid participant steer the time-window size")? Is it a precice parameter? Please refer to the parameter that makes this decision.

Reply:
I added the respective line in the config.